packages/frontend/src/components/DashboardFilter/Filter.tsx

```diff
@@ -169,68 +177,107 @@ const Filter: FC<Props> = ({
                         </Button>
                     </Tooltip>
                 ) : (
-                    <Button
-                        size="xs"
-                        variant={isTemporary ? 'outline' : 'default'}
-                        bg="white"
-                        rightIcon={
-                            (isEditMode || isTemporary) && (
-                                <CloseButton size="sm" onClick={onRemove} />
+                    <Indicator
+                        size={24}
+                        position="top-end"
+                        disabled={
```

Comment:
There's a logical error somewhere here. If I set the date type but not the actual value then the overlay disappears but the dashboard doesn't run.

Can we centralise this logic? They should be in sync.

![CleanShot 2024-03-27 at 10 13 33@2x](https://github.com/lightdash/lightdash/assets/11660098/c9fc2c6c-5d37-4785-9219-57b63984c360)

Reply:
Thanks for sharing this! 
Logic is centralised and fixed here: https://github.com/lightdash/lightdash/pull/9505/commits/c65f39cf8d788fe2681ed2db780bdf59510fdd97